### DISCUSSION TITLE
`reset` operator

### DISCUSSION THREAD

**Discussion by sergeysova** [sergeysova/patronum#178] [Ideas]
Sometimes we need to reset a bunch of stores on a single event, ex.: page unmounted

```ts
$post.reset(pageUnmounted)
$comments.reset(pageUnmounted)
$draftComment.reset(pageUnmounted)
```

I propose a new method `reset`:

```ts
reset({
  clock: pageUnmounted,
  target: [$post, $comments, $draftComment],
})
```

With support of multiple reset:

```ts
reset({
  clock: [pageUnmounted, pageMounted, $userSession],
  target: [$post, $comments, $draftComment],
})
```

**Comment by AlexandrHoroshih** [comment]
Sounds cool!

Implementation is obvious as well as the benefits 👍

**Comment by doasync** [comment]
I think it should work with a single target as well, so I would name the field `target` similar to the `sample`/`guard`.

**Reply by sergeysova** [reply]
But why you need it?

```ts
reset({ clock: event, target: $store })
// is the same as
$store.reset(event)
```

Second variant is shorter and already exists in effector

**Reply by AlexandrHoroshih** [reply]
I would agree here: for consistency with original operators of effector it would be cool to name field `target` + allow to pass only one store instead of array

This way it will be more clear, i think 🤔 
```ts
reset({
 clock: [pageUnmounted, reloadTriggered],
 target: $someState
})
```
if later there will be more stores to reset, you would just need to add more targets as array, just like with `sample`

**Reply by doasync** [reply]
> But why you need it?

So that you can use the full version everywhere (for consistency), without mixing the two styles, and avoid rewriting the shorter version into the full version when needed.

**Reply by sergeysova** [reply]
https://github.com/effector/patronum/pull/181

**Comment by sergeysova** [comment]
Locked, because already implemented
